src/pyobsplot/obsplot.py

```diff
@@ -319,3 +325,142 @@ def save_to_file(path: str, res: Union[SVG, HTML]) -> None:
+                self.render_typst(str(res.data), f.name)
+                width = spec["width"] if "width" in spec else 640
+                if "margin" in spec:
+                    width += 2*spec["margin"]
+                else: 
+                    if "marginLeft" in spec:
+                        width += spec["marginLeft"]
+                    if "marginRight" in spec:
+                        width += spec["marginRight"]
+                display(
+                    Image(filename=f.name, width=width, height=spec["height"] if "height" in spec else None)
+                )
+        else:
+            self.save_to_file(path, res)
+
+    def save_to_file(self, path: str, res: HTML) -> None:
+        if isinstance(path, io.StringIO):
+            path.write(str(res.data))
+            return
+        extension = Path(path).suffix.lower()
+        if extension not in [".png", ".jpg", ".jpeg", ".svg", ".pdf"]:
+            warnings.warn(
+                "Output file extension should be one of 'png', 'jpg', 'jpeg', 'svg', or 'pdf'",
+                RuntimeWarning,
+                stacklevel=1,
+            )
+        with open(path, "w", encoding="utf-8") as f:
+            self.render_typst(str(res.data), f.name)
+
+    @staticmethod
+    def shift_svg(svg):
+        soup = BeautifulSoup(str(svg), "xml")
+        svg = soup.svg
+        if "viewBox" in svg.attrs:
+            x, y, width, height = map(int, svg.attrs["viewBox"].split())
+            if x != 0 or y != 0:
+                g = soup.new_tag("g", transform=f"translate({-x}, {-y})")
+                g.extend(svg.contents)
+                svg.clear()
+                svg.append(g)
+                svg.attrs["viewBox"] = f"0 0 {width} {height}"
+        return str(svg)
+    
+    def render_typst(self, html: str, path: str) -> None:
+        path_obj = Path(path)
+        ext = "".join(path_obj.suffixes)
+        stem = str(path_obj.name).removesuffix("".join(path_obj.suffixes))
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            soup = BeautifulSoup(html, "xml")
+            figure = soup.find("figure", recursive=False)
+            swatches = []
+            plots = []
+            for i, swatch in enumerate(figure.find_all("div", recursive=False)):
+                new_swatch = []
+                for j, svg in enumerate(swatch.find_all("svg", recursive=True)):
+                    with open(f"{tmpdirname}/{stem}_{i}_{j}.svg", "w") as f:
+                        f.write(ObsplotTypstCreator.shift_svg(str(svg)))
+                    new_swatch.append(
+                        {"file": f"{stem}_{i}_{j}.svg", "width": svg.attrs["width"], "height": svg.attrs["height"], "text": svg.next_sibling}
+                    )
+                swatches.append(new_swatch)
+            for i, svg in enumerate(figure.find_all("svg", recursive=False)):
+                with open(f"{tmpdirname}/{stem}_{i}.svg", "w") as f:
```

Comment:
Without this change I get `UnicodeEncodeError: 'charmap' codec can't encode character '\u2192' in position 6270: character maps to <undefined>`

```suggestion
                with open(f"{tmpdirname}/{stem}_{i}.svg", "w", encoding = 'utf-8') as f:
```